### PR TITLE
Added improved profile functionality by giving users access to import…

### PR DESCRIPTION
**Pull request by gregoriofs**
…ant metrics and previous workouts that they can complete later while updating UI for homeviewcontroller. I also implemented notification functionality, though it does not seem to be working currently despite it functioning properly yesterday. Will make sure to correct the bug and expand notification functionality to all the desired use cases.

### CODE REVIEW

**Review by YitianLucyL — COMMENTED** [review]
In the PR description, you should include information:
1) summary of your project change i.e. what this PR want to achieve
2) your test plan i.e. screen recording, how did you ensure this diff achieve your goal

**Comment by YitianLucyL** [comment]
Thanks for removing the in-line logging!

File: Spotti/Spotti/View Controllers/ProfileViewController.m [hunk 1]
```diff
@@ -59,7 +54,44 @@ - (void)viewDidLoad {
...
+
+- (IBAction)didTapProfilePicture:(id)sender {
+    UIImagePickerController *imagePickerVC = [UIImagePickerController new];
+    imagePickerVC.delegate = self;
+    imagePickerVC.allowsEditing = YES;
+    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypeCamera;
+    }
+    else {
+        NSLog(@"Camera 🚫 available so we will use photo library instead");
```
Comment (YitianLucyL) [inline Spotti/Spotti/View Controllers/ProfileViewController.m]: Please remove 😄

File: Spotti/Spotti/View Controllers/ProfileViewController.m [hunk 2]
```diff
@@ -96,4 +128,41 @@ - (void)imagePickerController:(UIImagePickerController *)picker didFinishPicking
     [self dismissViewControllerAnimated:YES completion:nil];
 }
 
+- (void)getWorkouts{
+    PFQuery *query = [PFQuery queryWithClassName:@"Workout"];
+    [query whereKey:@"user" equalTo:[GymUser currentUser]];
+    [query setLimit:5];
+    [query orderByAscending:@"createdAt"];
+    [query findObjectsInBackgroundWithBlock:^(NSArray * _Nullable objects, NSError * _Nullable error) {
+            if(error != nil){
```
Comment (YitianLucyL) [inline Spotti/Spotti/View Controllers/ProfileViewController.m:138]: Nice error handling!

File: Spotti/Spotti/Views/ExerciseCell.m [hunk 3]
```diff
@@ -19,4 +19,6 @@ - (void)setSelected:(BOOL)selected animated:(BOOL)animated {
     // Configure the view for the selected state
 }
 
+
+
```
Comment (YitianLucyL) [inline Spotti/Spotti/Views/ExerciseCell.m:23]: why adding two blank lines here

File: Spotti/Spotti/View Controllers/ProfileViewController.m [hunk 4]
```diff
@@ -96,4 +128,41 @@ - (void)imagePickerController:(UIImagePickerController *)picker didFinishPicking
...
+                NSLog(@"%@", error.localizedDescription);
+            }
+            else{
+                self.arrayOfWorkouts = objects;
+                [self.workoutTableView reloadData];
+            }
+    }];
```
Comment (YitianLucyL) [inline Spotti/Spotti/View Controllers/ProfileViewController.m:145]: Remove in-line logging before commit!

**Review by YitianLucyL — APPROVED** [review]
Overall LGTM! Please remove in-line logging and unnecessary blank lines

**Comment by gregoriofs** [comment]
I improved how workouts looked so they'd properly display previously inputted sets and reps and added displaying of images for exercises, though only with placeholder image for now, furthering the workout generation functionality.
ezgif com-gif-maker (8)

**Pull request merged**